[DatePipe] add deprecation warnings for DataPipes that will solely exist in TorchData

NivekT opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65827

Differential Revision: [D31272794](https://our.internmc.facebook.com/intern/diff/D31272794)

cc @VitalyFedyunin @ejguan @NivekT

NivekT committed:
[DatePipe] add deprecation warnings for DataPipes that will solely exist in TorchData

[ghstack-poisoned]

NivekT committed:
Update on "[DatePipe] add deprecation warnings for DataPipes that will solely exist in TorchData"




[ghstack-poisoned]

NivekT commented:
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31272794).

NivekT committed:
Update on "[DatePipe] add deprecation warnings for DataPipes that will solely exist in TorchData"


Differential Revision: [D31272794](https://our.internmc.facebook.com/intern/diff/D31272794)

cc @VitalyFedyunin ejguan @NivekT

[ghstack-poisoned]

ejguan commented on torch/utils/data/datapipes/utils/common.py:
```diff
@@ -63,3 +63,8 @@ def validate_pathname_binary_tuple(data):
+# Warns user that the DataPipe has been moved to TorchData and will be removed from `torch`
+def deprecation_warning_torchdata(name):
+    warnings.warn(f"""{name} is deprecated and will be removed from the package `torch`. Please import it """
+                   """from the new package TorchData: https://github.com/pytorch/data""", DeprecationWarning)
```
nit: Maybe also talk about functional API for these DataPipe also exists in TorchData.

NivekT commented:
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31272794).

ejguan reviewed (approved):
LGTM

NivekT committed:
Update on "[DatePipe] add deprecation warnings for DataPipes that will solely exist in TorchData"


Differential Revision: [D31272794](https://our.internmc.facebook.com/intern/diff/D31272794)

cc @VitalyFedyunin ejguan @NivekT

[ghstack-poisoned]

NivekT commented:
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31272794).

ejguan commented:
Same as the another PR. Could we add this PR to release 1.10?
